providers/src/airflow/providers/common/compat/lineage/hook.py

```diff
@@ -16,7 +16,10 @@
 # under the License.
 from __future__ import annotations
 
-from airflow.providers.common.compat import AIRFLOW_V_2_10_PLUS, AIRFLOW_V_3_0_PLUS
+from airflow.providers.common.compat import (
+    PROVIDERS_COMMON_COMPAT_AIRFLOW_IS_2_10_PLUS,
+    PROVIDERS_COMMON_COMPAT_AIRFLOW_IS_3_0_PLUS,
```

Comment:
Why did we change the constant name?

Reply:
Because this already happened that they were imported from different (and wrong places) - these constants were same named in 3 places and there were few mistakes where they were imported from wrong places.

Reply:
Many times (and that already happened) people just do "Ctrl-Shift-O" and constants are imported from wrong place and reviewers are missing it (happened already) - renaming those constants makes it much more difficult to make the mistake (for example importing same constant from different provider) when you are reviewing the code.

That's the root cause that already happened in the past and this PR already solved one of those that were left and missed.

Reply:
https://docs.astral.sh/ruff/rules/banned-api/ then.

Reply:
No. Using this you (unless you find a different way) you have to implement 90+ rules to ban imports from N-1 providers - different for each provider sub tree. Surely we could generate those rules (again by pre-commit probably).

But maybe I am mistaken and there is an easy way to use that ruff rule to ban "every constant name this way from any other package except import of it from my own subtree"

Do you think it is possible @ashb?

Reply:
> That's the root cause that already happened in the past and this PR already solved one of those that were left and missed

The issue that started this change seemed to be about manual changes to generated files, and preventing that is un-contentious. But it them seemed to transform into "lets build a load of scaffolding to be unified about version checks". That's the part I have problems with as this solution makes everything incrementally more complex, each individual change is "yeah okay, slightly complex" but in aggregate all this pre-commit+autogenerated code+magic leads to a hugely complex system.

That said listening to an episode of Talk Python To Me this morning https://import-linter.readthedocs.io/en/stable/index.html came up. I'm trying this quickly to see if we can make it not allow imports from a provider it doesn't list as a dep on.

Reply:
Yes. Ruff analyse graph seems to do the job nicely.